test/e2e/auth/subjectreviews.go

```diff
@@ -0,0 +1,156 @@
+			UserName: saUsername,
+			UID:      saUID,
+			Groups:   saGroups,
+		}
+
+		impersonatedClientSet, err := kubernetes.NewForConfig(config)
+		framework.ExpectNoError(err, "Could not load config, %v", err)
+
+		ginkgo.By(fmt.Sprintf("Creating SubjectAccessReview for %q", saUsername))
+
+		sar := &authorizationv1.SubjectAccessReview{
+			Spec: authorizationv1.SubjectAccessReviewSpec{
+				ResourceAttributes: &authorizationv1.ResourceAttributes{
+					Verb:      "list",
+					Resource:  "configmaps",
+					Namespace: ns,
+					Version:   "v1",
+				},
+				User:   saUsername,
+				Groups: saGroups,
+				UID:    saUID,
+			},
+		}
+
+		sarResponse, err := AuthClient.SubjectAccessReviews().Create(context.TODO(), sar, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "Unable to create a SubjectAccessReview, %#v", err)
+		framework.Logf("sarResponse Status: %#v", sarResponse.Status)
+		expectedAllowed := sarResponse.Status.Allowed
+
+		ginkgo.By(fmt.Sprintf("Verifying as %q api 'list' configmaps in %q namespace", saUsername, ns))
+		_, err = impersonatedClientSet.CoreV1().ConfigMaps(ns).List(context.TODO(), metav1.ListOptions{})
+
+		actuallyAllowed := false
+		switch {
+		case apierrors.IsForbidden(err):
+			actuallyAllowed = false
+		case err != nil:
+			framework.Fail("Unexpected error")
+		default:
+			actuallyAllowed = true
+		}
+
+		if actuallyAllowed != expectedAllowed {
+			framework.Fail(fmt.Sprintf("Could not verify SubjectAccessReview for %q in namespace %q", saUsername, ns))
+		}
+		framework.Logf("SubjectAccessReview has been verified")
+
+		ginkgo.By(fmt.Sprintf("Creating a LocalSubjectAccessReview for %q", saUsername))
+
+		lsar := &authorizationv1.LocalSubjectAccessReview{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: ns,
+			},
+			Spec: authorizationv1.SubjectAccessReviewSpec{
+				ResourceAttributes: &authorizationv1.ResourceAttributes{
+					Verb:      "list",
+					Resource:  "configmaps",
+					Namespace: ns,
+					Version:   "v1",
+				},
+				User:   saName,
+				Groups: saGroups,
+				UID:    saUID,
+			},
+		}
+
+		lsarResponse, err := AuthClient.LocalSubjectAccessReviews(ns).Create(context.TODO(), lsar, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "Unable to create a LocalSubjectAccessReview, %#v", err)
+		framework.Logf("lsarResponse Status: %#v", lsarResponse.Status)
+		expectedAllowed = lsarResponse.Status.Allowed
+
+		ginkgo.By(fmt.Sprintf("Verifying as %q api 'list' configmaps in %q namespace", saUsername, ns))
+		_, err = impersonatedClientSet.CoreV1().ConfigMaps(ns).List(context.TODO(), metav1.ListOptions{})
+
+		actuallyAllowed = false
+		switch {
+		case apierrors.IsForbidden(err):
+			actuallyAllowed = false
+		case err != nil:
+			framework.Fail("Unexpected error")
+		default:
+			actuallyAllowed = true
+		}
+
+		if actuallyAllowed != expectedAllowed {
+			framework.Fail(fmt.Sprintf("Could not verify LocalSubjectAccessReview for %q in namespace %q", saUsername, ns))
```

Comment:
if this fails, more info will be helpful in debugging:

```suggestion
			framework.Fail(fmt.Sprintf("Could not verify LocalSubjectAccessReview for %q in namespace %q: LocalSubjectAccessReview allowed: %v, request allowed: %v, request error: %v", saUsername, ns, expectedAllowed, actuallyAllowed, requestErr))
```